grpccache.go

```diff
@@ -164,6 +169,31 @@ func (c *Cache) Store(ctx context.Context, method string, arg proto.Message, res
 	return nil
 }
 
+// cleanOldEntries removes expired entries from the cache. It does so in a
+// separate goroutine and only once every 10 seconds. The method assumes that
+// the cache mutex is locked.
+func (c *Cache) cleanOldEntries() {
+	now := time.Now()
+	if now.After(c.lastClean.Add(10 * time.Second)) {
+		c.lastClean = now
+		// clear old entries asynchronously
+		go func() {
+			c.mu.Lock()
```

Comment:
This holds a lock while it iterates through all cache entries, which would block every single gRPC request, right? This seems like a cheap operation, but should we upper bound the time somehow?

Also, cache exhaustion is still a possibility here. We should implement an actual eviction policy in the future. @neelance can you file an issue to track on the Trello? You can put it under tech debt and we'll make sure it's prioritized.